lib/SIL/SILFunctionType.cpp

```diff
@@ -2259,15 +2259,31 @@ static CanType copyOptionalityFromDerivedToBase(TypeConverter &tc,
   // (T1 -> T2) +> (S1 -> S2) = (T1 +> S1) -> (T2 +> S2)
   if (auto derivedFunc = dyn_cast<AnyFunctionType>(derived)) {
     if (auto baseFunc = dyn_cast<AnyFunctionType>(base)) {
-      return CanAnyFunctionType::getOld(
-        baseFunc.getOptGenericSignature(),
-        copyOptionalityFromDerivedToBase(tc,
-                                         derivedFunc.getInput(),
-                                         baseFunc.getInput()),
-        copyOptionalityFromDerivedToBase(tc,
-                                         derivedFunc.getResult(),
-                                         baseFunc.getResult()),
-        baseFunc->getExtInfo());
+      SmallVector<FunctionType::Param, 8> params;
+
+      auto derivedParams = derivedFunc.getParams();
+      auto baseParams = baseFunc.getParams();
+      assert(derivedParams.size() == baseParams.size());
+      for (unsigned i = 0, e = derivedParams.size(); i < e; i++) {
+        // FIXME: Why are 'escaping' flags set inconsistently?
+        assert(derivedParams[i].getParameterFlags().withEscaping(false) ==
```

Comment:
`escaping` in the parameter flags is just for AST printing.  I wouldn't worry about it with an assert.